src/GprTool/NuGetUtilities.cs

```diff
@@ -1,11 +1,186 @@
+        {
+            if (packageFile == null) throw new ArgumentNullException(nameof(packageFile));
+            
+            var randomId = Guid.NewGuid().ToString("N");
+  
+            using var packageArchiveReader = new PackageArchiveReader(
+                packageFile.FilenameAbsolutePath.ReadSharedToStream(), false);
+            
+            var nuspecXDocument = packageArchiveReader.NuspecReader.Xml;
+            var packageXElement = nuspecXDocument.Single("package");
+            var metadataXElement = packageXElement.Single("metadata");
+            var packageId = packageXElement.Single("id").Value;
+            var versionXElement = metadataXElement.Single("version");
+            
+            if (nuGetVersion != null)
+            {
+                versionXElement.SetValue(nuGetVersion); 
+            }
+            else
+            {
+                nuGetVersion = NuGetVersion.Parse(versionXElement.Value);
+            }
+
+            var repositoryXElement = metadataXElement.SingleOrDefault("repository");
+            if (repositoryXElement == null)
+            {
+                repositoryXElement = new XElement("repository");
+                repositoryXElement.SetAttributeValue("url", packageFile.RepositoryUrl);
+                repositoryXElement.SetAttributeValue("type", "git");
+                metadataXElement.Add(repositoryXElement);
+            }
+            else
+            {
+                repositoryXElement.SetAttributeValue("url", packageFile.RepositoryUrl);
+                repositoryXElement.SetAttributeValue("type", "git");
+            }
+            
+            using var nuspecMemoryStream = new MemoryStream();
+            nuspecXDocument.Save(nuspecMemoryStream);
+            nuspecMemoryStream.Seek(0, SeekOrigin.Begin);
+
+            var packageFileWorkingDirectoryAbsolutePath = Path.GetDirectoryName(packageFile.FilenameAbsolutePath);
+            var packageFileRewriteWorkingDirectory = Path.Combine(packageFileWorkingDirectoryAbsolutePath,
+                $"{packageId}.{nuGetVersion}_{randomId}");
+
+            using var tmpDirectory = new DisposableDirectory(packageFileRewriteWorkingDirectory);
+
+            ZipFile.ExtractToDirectory(packageFile.FilenameAbsolutePath, tmpDirectory);
+
+            var nuspecDstFilename = Path.Combine(tmpDirectory, $"{packageId}.nuspec");
+            File.WriteAllBytes(nuspecDstFilename, nuspecMemoryStream.ToArray());
+
+            using var outputStream = new MemoryStream();
+            
+            var packageBuilder = new PackageBuilder(nuspecMemoryStream, tmpDirectory,
+                propertyProvider => throw new NotImplementedException());
+            packageBuilder.Save(outputStream);
+
+            packageFile.FilenameAbsolutePath = Path.Combine(packageFileWorkingDirectoryAbsolutePath,
+                $"{packageId}.{nuGetVersion}_{randomId}_gpr.nupkg");
```

Comment:
Could we keep this simple and make it something like this?

```suggestion
            packageFile.FilenameAbsolutePath = $"{packageFile.FilenameAbsolutePath}.zip");
```

Reply:
Fixed in https://github.com/jcansdale/gpr/pull/53/commits/47f04097cc3357e00bb8cbcd4da01648d8400de1